common/Models/ExtensionAdaptiveCardSession.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DevHome.Common.Helpers;
+
+using Microsoft.Windows.DevHome.SDK;
+using Serilog;
+using Windows.Foundation;
+
+namespace DevHome.Common.Models;
+
+/// <summary>
+/// Wrapper class for the IExtensionAdaptiveCardSession and IExtensionAdaptiveCardSession2 interfaces.
+/// </summary>
+public class ExtensionAdaptiveCardSession
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(ExtensionAdaptiveCardSession));
+
+    private readonly string _componentName = "ExtensionAdaptiveCardSession";
+
+    public IExtensionAdaptiveCardSession Session { get; private set; }
+
+    public event TypedEventHandler<ExtensionAdaptiveCardSession, ExtensionAdaptiveCardSessionStoppedEventArgs>? Stopped;
+
+    public ExtensionAdaptiveCardSession(IExtensionAdaptiveCardSession cardSession)
+    {
+        Session = cardSession;
+
+        if (Session is IExtensionAdaptiveCardSession2 cardSession2)
+        {
+            cardSession2.Stopped += OnSessionStopped;
+        }
+    }
+
+    public ProviderOperationResult Initialize(IExtensionAdaptiveCard extensionUI)
+    {
+        try
+        {
+            return Session.Initialize(extensionUI);
+        }
+        catch (Exception ex)
+        {
+            _log.Error(_componentName, $"Initialize failed due to exception", ex);
+            return new ProviderOperationResult(ProviderOperationStatus.Failure, ex, ex.Message, ex.Message);
+        }
+    }
+
+    public void Dispose()
+    {
+        try
+        {
+            Session.Dispose();
+        }
+        catch (Exception ex)
+        {
+           _log.Error(_componentName, $"Dispose failed due to exception", ex);
+        }
+
+        if (Session is IExtensionAdaptiveCardSession2 cardSession2)
```

Comment:
This doesn't seem right. Session was just disposed above and now we are making a call into it. Even though it's only unsubscribing from an event we shouldn't reason about Session internal implementation whether it can handle that call or not after disposing it.

Reply:
yea I'll swap the order to unsubscribe first, then dispose second